test/specs/test-error-parsers.el

```diff
@@ -0,0 +1,99 @@
+;;; test-code-style.el --- Flycheck Specs: Error parsers     -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2016 Sebastian Wiesner and Flycheck contributors
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; Specs for generic error parsers.
+
+;;; Code:
+
+(require 'flycheck-buttercup)
+
+(defconst flycheck/checkstyle-xml
+  "<?xml version=\"1.0\" encoding=\"utf-8\"?>
+<checkstyle version=\"4.3\">
+  <file name=\"test-javascript/missing-semicolon.js\">
+    <error line=\"3\" column=\"21\" severity=\"error\"
+           message=\"Missing semicolon.\" source=\"Foo3\" />
+    <error line=\"3\" severity=\"warning\"
+           message=\"Implied global &apos;alert&apos;\" source=\"Foo4\" />
+  </file>
+  <file name=\"test-javascript/missing-quote.js\">
+    <error line=\"1\" column=\"undefined\" severity=\"error\"
+           message=\"Cannot read property &apos;id&apos; of undefined\"
+           source=\"Foo1\" />
+  </file>
+</checkstyle>"
+  "Example Checkstyle output from jshint.")
+
+(defconst flycheck/checkstyle-expected-errors
+  (list
+   (flycheck-error-new
+    :filename "test-javascript/missing-semicolon.js"
+    :checker 'checker
+    :buffer 'buffer
+    :line 3
+    :column 21
+    :level 'error
+    :message "Missing semicolon."
+    :id "Foo3")
+   (flycheck-error-new
+    :filename "test-javascript/missing-semicolon.js"
+    :checker 'checker
+    :buffer 'buffer
+    :line 3
+    :column nil
+    :level 'warning
+    :message "Implied global 'alert'"
+    :id "Foo4")
+   (flycheck-error-new
+    :filename "test-javascript/missing-quote.js"
+    :checker 'checker
+    :buffer 'buffer
+    :line 1
+    :column nil
+    :level 'error
+    :message "Cannot read property 'id' of undefined"
+    :id "Foo1"))
+  "Errors to be parsed from `flycheck-checkstyle-xml'.")
+
+(describe "Error parsers"
+  (describe "Checkstyle XML parsing"
```

Comment:
Use "The checkstyle parser" as argument for `describe`.  That reads better in the test output imho.